test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +105,63 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', {
+        data: readable,
+        dataType: DATA_TYPE_BINARY,
+        metadata: { raw: [], linkRel: { type: new Set() }, profiles: []} as RepresentationMetadata,
+      })));
+
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const waitForMe = new Promise((resolve): any => {
+      representation.data.on('data', (): any => readable);
+      representation.data.prependListener('end', (): any => {
+        if (!order.includes('end')) {
+          order.push('end');
+        }
+        resolve();
+        return readable;
+      });
+    });
+    await waitForMe;
+    expect(locker.acquire).toHaveBeenCalledTimes(1);
+    expect(locker.acquire).toHaveBeenLastCalledWith({ path: 'path' });
+    expect(source.getRepresentation).toHaveBeenCalledTimes(1);
+    expect(lock.release).toHaveBeenCalledTimes(1);
+    expect(order).toEqual([ 'acquire', 'getRepresentation', 'end', 'release' ]);
+  });
+
+  it('releases the lock on the resource when readable errors.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', {
+        data: readable,
+        dataType: DATA_TYPE_BINARY,
+        metadata: { raw: [], linkRel: { type: new Set() }, profiles: []} as RepresentationMetadata,
+      })));
+
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const waitForMe = new Promise((resolve): any => {
+      representation.data.on('data', (): any => readable);
+      representation.data.prependListener('error', (): any => {
+        if (!order.includes('error')) {
+          order.push('error');
+        }
+        resolve();
+        return readable;
+      });
+    });
```

Comment:
Since an error is going to be triggered by `destroy` this Promise isn't necessary. Listening to `'error'` is still required but doesn't have to be in a Promise.

Reply:
If I remove the Promise wrap, jest complains `lock.release` isn't called.

Reply:
Ah yes, the issue is that jest checks immediately if there was a call, while this call only happens in a callback. But you can still remove the data drain from the promise then, only listening to the error is necessary.